supertokens_python/session/middleware.py

```diff
@@ -0,0 +1,47 @@
+"""
+Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+
+This software is licensed under the Apache License, Version 2.0 (the
+"License") as published by the Apache Software Foundation.
+
+You may not use this file except in compliance with the License. You may
+obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations
+under the License.
+"""
+from __future__ import annotations
+from typing import Union, TYPE_CHECKING
+
+from supertokens_python.framework.index import FRAMEWORKS
+from supertokens_python.utils import normalise_http_method
+
+if TYPE_CHECKING:
+    from supertokens_python.framework.request import BaseRequest
+    from .session_class import Session
+    from .session_recipe import SessionRecipe
+from supertokens_python.normalised_url_path import NormalisedURLPath
+
+
+def verify_session(recipe: SessionRecipe, anti_csrf_check: Union[bool, None] = None, session_required: bool = True):
```

Comment:
are you sure this function will not be required to be implemented for each framework? Maybe some wrapper might be required just like middleware. I'm not sure